autonomy/cli/helpers/deployment.py

```diff
@@ -163,6 +168,127 @@ def run_deployment(
         stop_deployment(build_dir=build_dir)
 
 
+def _prepare_agent_env(working_dir: Path) -> None:
+    """Prepare agent env, add keys, run aea commands."""
+    env = json.loads((working_dir / "agent.json").read_text(encoding="utf-8"))
+    # Patch for trader agent
+    if "SKILL_TRADER_ABCI_MODELS_PARAMS_ARGS_STORE_PATH" in env:
+        data_dir = working_dir / "data"
+        data_dir.mkdir(exist_ok=True)
+        env["SKILL_TRADER_ABCI_MODELS_PARAMS_ARGS_STORE_PATH"] = str(data_dir)
+
+    # TODO: Dynamic port allocation, backport to service builder
+    env["CONNECTION_ABCI_CONFIG_HOST"] = "localhost"
+    env["CONNECTION_ABCI_CONFIG_PORT"] = "26658"
+
+    for var in env:
+        # Fix tendermint connection params
+        if var.endswith("MODELS_PARAMS_ARGS_TENDERMINT_COM_URL"):
+            env[var] = "http://localhost:8080"
+
+        if var.endswith("MODELS_PARAMS_ARGS_TENDERMINT_URL"):
+            env[var] = "http://localhost:26657"
+
+        if var.endswith("MODELS_PARAMS_ARGS_TENDERMINT_P2P_URL"):
+            env[var] = "localhost:26656"
```

Comment:
What if those ports are occupied? I am not sure why we hardcode these and do not set the ones provided in the configuration.

One more thing to discuss with the operate team.

Reply:
We keep the same assumptions as with the docker and kube builds, i.e., the builds should work no matter the values in the configurations. We will keep this part and will also implement dynamic port allocation in the future (#2259).